core/templates/dev/head/pages/exploration_editor/editor_tab/customize_interaction_modal_directive.html

```diff
@@ -67,3 +66,9 @@ <h3 ng-if="stateInteractionIdService.displayed && (customizationModalReopened ||
     Okay
   </button>
 </div>
+<style>
+  .interaction-editor-form .boolean-checkbox {
```

Comment:
Can you prefix all this with the name of the directive (customize-interaction-modal)? That makes sure the style is scoped to this component only.

@apb7 -- FYI, this might be something that might be worth automating checks for.

Reply:
I think we can not name that,because the the html is injected to the body of the modal through template(path to customize_interaction_modal_directive.html). so,there is no <customize -interaction-modal> tag to name in that way.so I added class name for the form for scoping style to that component only.

Reply:
Could you please paste here the part of the DOM tree containing this entire component? You can get it by doing "Inspect Element" in your browser and looking at the Elements tab.

Reply:
`<div uib-modal-window="modal-window" class="modal fade in" role="dialog" index="0" animate="animate" ng-style="{'z-index': 1050 + $$topModalIndex*10, display: 'block'}" tabindex="-1" uib-modal-animation-class="fade" modal-in-class="in" modal-animation="true" style="z-index: 1050; display: block;"><div class="modal-dialog "><div class="modal-content" uib-modal-transclude=""><div class="modal-header" style="border-bottom: 0; padding-bottom: 0;">
  <!---->

  <!----><h3 ng-if="stateInteractionIdService.displayed &amp;&amp; (customizationModalReopened || hasCustomizationArgs)" class="" style="">
    Customize Interaction (Fraction Input)
  </h3><!---->
</div>

<div class="modal-body">
  <!---->

  <!----><form ng-if="stateInteractionIdService.displayed &amp;&amp; hasCustomizationArgs" name="form.schemaForm" class="interaction-editor-form protractor-test-interaction-editor ng-pristine ng-valid" style="">
    <!----><div ng-repeat="customizationArgSpec in customizationArgSpecs track by $index">
      <!---->
        <schema-based-editor ng-class="{'boolean-checkbox': customizationArgSpec.schema.type === 'bool'}" local-value="stateCustomizationArgsService.displayed[customizationArgSpec.name].value" schema="customizationArgSpec.schema" class="boolean-checkbox"><!---->

<!----><schema-based-bool-editor ng-if="schema().type === 'bool' &amp;&amp; !schema().choices" local-value="$parent.localValue" is-disabled="isDisabled()" label-for-focus-target="labelForFocusTarget()"><div class="input-group">
  <!----><span ng-if="!expressionMode">
    <input type="checkbox" ng-model="$parent.localValue" ng-disabled="isDisabled()" focus-on="" class="ng-pristine ng-untouched ng-valid ng-empty" aria-invalid="false">
  </span><!---->
</div>
</schema-based-bool-editor><!---->

<!---->

<!---->

<!---->

<!---->

<!---->

<!---->

<!---->
</schema-based-editor>
      <!----><div ng-if="customizationArgSpec.schema.type === 'bool'" class="oppia-interaction-customization-label">Require the learner's answer to be in simplest form</div><!---->
      <div style="margin-bottom: 15px;"></div>
    </div><!----><div ng-repeat="customizationArgSpec in customizationArgSpecs track by $index">
      <!---->
        <schema-based-editor ng-class="{'boolean-checkbox': customizationArgSpec.schema.type === 'bool'}" local-value="stateCustomizationArgsService.displayed[customizationArgSpec.name].value" schema="customizationArgSpec.schema" class="boolean-checkbox"><!---->

<!----><schema-based-bool-editor ng-if="schema().type === 'bool' &amp;&amp; !schema().choices" local-value="$parent.localValue" is-disabled="isDisabled()" label-for-focus-target="labelForFocusTarget()"><div class="input-group">
  <!----><span ng-if="!expressionMode">
    <input type="checkbox" ng-model="$parent.localValue" ng-disabled="isDisabled()" focus-on="" class="ng-pristine ng-untouched ng-valid ng-not-empty" aria-invalid="false">
  </span><!---->
</div>
</schema-based-bool-editor><!---->

<!---->

<!---->

<!---->

<!---->

<!---->

<!---->

<!---->
</schema-based-editor>
      <!----><div ng-if="customizationArgSpec.schema.type === 'bool'" class="oppia-interaction-customization-label">Allow improper fractions in the learner's answer</div><!---->
      <div style="margin-bottom: 15px;"></div>
    </div><!----><div ng-repeat="customizationArgSpec in customizationArgSpecs track by $index">
      <!---->
        <schema-based-editor ng-class="{'boolean-checkbox': customizationArgSpec.schema.type === 'bool'}" local-value="stateCustomizationArgsService.displayed[customizationArgSpec.name].value" schema="customizationArgSpec.schema" class="boolean-checkbox"><!---->

<!----><schema-based-bool-editor ng-if="schema().type === 'bool' &amp;&amp; !schema().choices" local-value="$parent.localValue" is-disabled="isDisabled()" label-for-focus-target="labelForFocusTarget()"><div class="input-group">
  <!----><span ng-if="!expressionMode">
    <input type="checkbox" ng-model="$parent.localValue" ng-disabled="isDisabled()" focus-on="" class="ng-pristine ng-untouched ng-valid ng-not-empty" aria-invalid="false">
  </span><!---->
</div>
</schema-based-bool-editor><!---->

<!---->

<!---->

<!---->

<!---->

<!---->

<!---->

<!---->
</schema-based-editor>
      <!----><div ng-if="customizationArgSpec.schema.type === 'bool'" class="oppia-interaction-customization-label">Allow the answer to contain an integer part</div><!---->
      <div style="margin-bottom: 15px;"></div>
    </div><!----><div ng-repeat="customizationArgSpec in customizationArgSpecs track by $index">
      <!----><div ng-if="customizationArgSpec.schema.type !== 'bool'" class="oppia-interaction-customization-label">Custom placeholder text (optional)</div><!---->
        <schema-based-editor ng-class="{'boolean-checkbox': customizationArgSpec.schema.type === 'bool'}" local-value="stateCustomizationArgsService.displayed[customizationArgSpec.name].value" schema="customizationArgSpec.schema"><!---->

<!---->

<!---->

<!---->

<!----><schema-based-unicode-editor ng-if="schema().type === 'unicode' &amp;&amp; !schema().choices" local-value="$parent.localValue" is-disabled="isDisabled()" validators="schema().validators" ui-config="schema().ui_config" label-for-focus-target="labelForFocusTarget()" on-input-blur="onInputBlur" on-input-focus="onInputFocus"><span>
  <!---->

  <!----><span ng-if="!getCodingMode() &amp;&amp; !getRows()" class="" style="">
    <input type="text" ng-disabled="isDisabled()" class="form-control ng-pristine ng-untouched ng-valid ng-empty" ng-model="$parent.localValue" ng-blur="onInputBlur()" ng-focus="onInputFocus()" focus-on="" ng-keypress="onKeypress($event)" placeholder="" apply-validation="validators()" style="display: inline;" aria-invalid="false">
  </span><!---->

  <!-- Note that we do not yet support ng-blur for the code editor. -->
  <!---->
</span>
</schema-based-unicode-editor><!---->

<!---->

<!---->

<!---->

<!---->
</schema-based-editor>
      <!---->
      <div style="margin-bottom: 15px;"></div>
    </div><!---->
  </form><!---->

  <!---->
</div>

<div class="modal-footer">
  <!---->
  <!-- Do not show this button if an interaction is already selected. -->
  <!----><button class="btn btn-default" style="float: left;" ng-click="returnToInteractionSelector()" ng-if="stateInteractionIdService.displayed &amp;&amp; !stateInteractionIdService.savedMemento &amp;&amp; hasCustomizationArgs">Change Interaction</button><!---->

  <!----><button class="btn btn-default" ng-click="cancel()" ng-if="hasCustomizationArgs">Cancel</button><!---->
  <!----><div class="btn-group" ng-if="hasCustomizationArgs" popover-append-to-body="" popover-placement="bottom" popover-trigger="none" ng-attr-uib-popover="<[getSaveInteractionButtonTooltip()]>" uib-popover="" style="">
    <button class="btn btn-success protractor-test-save-interaction" ng-click="save()" ng-disabled="!isSaveInteractionButtonEnabled()">
      Save Interaction
    </button>
  </div><!---->
  <!---->
</div>
<style>
  .interaction-editor-form .boolean-checkbox {
    float: left;
    margin-right: 4px;
  }
</style>
</div></div>
</div>`

Reply:
Ah, I see, it's transcluded. OK, thanks! I think this looks good from a code perspective.

Once you get a UI review from @prasanna08 I think you're all set!

Reply:
thanks for the review!